Modules/_multiprocessing/semaphore.c

```diff
@@ -780,6 +778,7 @@ PyTypeObject _PyMp_SemLockType = {
     /* tp_init           */ 0,
     /* tp_alloc          */ 0,
     /* tp_new            */ _multiprocessing_SemLock,
+    .tp_free = PyObject_Free,
```

Comment:
I may be missing something but I don't think you need this. The default for the static type case should be `PyObject_Del` which is equivalent to `PyObject_Free` and inheritance will replace it for the appropriate generic alloc deallocator.

Reply:
Okay, I wanted to make it explicit for others which may not know this but I am fine with removing it. removed.